### ISSUE TITLE
Setting 'Minimize window at application startup' don't work as expected 

### ISSUE THREAD

**Issue by railes** [keepassxreboot/keepassxc#1147]
## Expected Behavior
KeePassXC should ask on user login for password to unlock database and then minimize and hide to tray

## Current Behavior
KeePassXC asks on user login for password to unlock database, but window stays open after password entry

## Possible Solution
KeePassXC should determine on start:
1. if password is needed, open window to enter password (works)
2. after password entry determine from settings if KeePassXC should 'Minimize window at application startup' and 'Hide window to system tray when minimized' (don't work)

## Steps to Reproduce (for bugs)
customize KeePassXC:
1. set KeePassXC to be started on user login (KDE autostart settings)
2. start KeePassXc manually
3. Check option 'Minimize window at application startup'
4. Check option 'Hide window to system tray when minimized'
5. Check option 'Hide window to system tray instead of App Exit'
6. Save the settings and quit KeePassXC 

testing:
1. either start KeePassXC manually or relogin user
(on relogin window to ask for password popups automatically, on manual start klick on system tray app) 
2. enter password and hit enter
3. window with database entries stays open (should not be like this)

## Context

This is a minor Bug (workaround: minizime window manually)
But it's still annoying to have a feature which doesn't work as expected ;-)

## Debug Info
KeePassXC - Version 2.2.2
Revision: 6d46717

Libraries:
- Qt 5.6.2
- libgcrypt 1.6.1

Operating system: openSUSE Leap 42.3
CPU architecture: x86_64
Kernel: linux 4.4.92-31-default

Enabled extensions:
- KeePassHTTP
- Auto-Type
- YubiKey

KDE Plasma Version: 5.8.7
KDE Frameworks version: 5.32.0 
RPM version: https://download.opensuse.org/repositories/security/openSUSE_Leap_42.3/x86_64/keepassxc-2.2.2-17.1.x86_64.rpm

**Comment by phoerious** [comment]
This looks like it's related to #155, which is a WONTFIX. The current KeePassXC behavior is to minimize to tray and only ask to unlock if the user actively brings the application to the foreground.
What you describe, though, sounds like the opposite. I'm not sure how you observed that behavior. When the application is started, it stays hidden in the tray until you click on the icon and then asks you to enter the password. That is expected behavior and not a bug.

**Comment by railes** [comment]
Yes phoerious, you are right, I haven't seen #155.

When I customize KeePassXC with 'Minimize window at application startup' and 'Hide window to system tray when minimized', it still pops-up when I login as a user in KDE and asks me for the password (what I expected).
But when I've entered the password (and hit enter), it don't go minimizing and hiding to tray (I expected this too).

The point that an application goes 'hiding' in tray is waiting for an action in background. And that's what KeePassXC should do.

A pop-up or an open window is only necessary in case of
- entering the password at login (which works)
- entering the password on unlocking the database, e.g. again after 'pausing' a session or closing lid (works not, I've to open the password dialogue manually by clicking on symbol in tray, this is acceptable since KeePassXC asks for the password at next Auto-Type)
- on demand by clicking on symbol in tray with opening 'main window' to administrate the database (which works)
- on asking to select an entry to Auto-Type (which works)

But in the meantime KeePassXC should wait in tray after entering the password (works not, I've to close manually the 'main window' after entering the password).

For my opinion the solution for bug 'Start minimized first asks to unlock the DB' #105 contradicts the behaviour of a 'waiting' application (an not a 'sleeping' unlocked application).

But I could also accept the actual behaviour of KeePassXC, but then the application should not ask for a password at login!? which it does (in my case under KDE)...
But I can also live with the actual behaviour and if you like put WONTFIX on my demand ;-)

**Comment by phoerious** [comment]
I cannot reproduce that KeePassXC pops up when starting KDE. For me it starts hidden in the tray, which is expected.
The behavior you describe as desired (pops up, asks for password and then hides after it was entered), was a bug (see #105) which was fixed a long time ago.

**Comment by railes** [comment]
OK, as I said please put WONTFIX on this
the straneg behavoiur of KDE is not a problem for me

**Comment by adomas187** [comment]
I have same issue, is there some kind of workaround for this? Would be really good to know, since I could not solve this.

KDE 5.40.0 / Plasma 5.11.3

**Comment by railes** [comment]
Hi,
my 'workaround' is
- give password for database directly after KDE login in KeePassXC popup window 
- close KeePassXC popup window to tray (e.g. with ALT+F4)

I noticed a strange behaviour:
Plasma5 is quite slow in creating the desktop, so I've to wait on a black screen some time (>10sec) to see the desktop theme and panel coming on
I don't know the reason for this, but has nothing to do with KeePassXC, maybe since I've four activities or whatever it is
but the popup window of KeePassXC is quicker than the rest, so I see the popup window first (and no theme or panel)!?

**Comment by raphwriter** [comment]
I solved/worked around this issue by using KWins Window Rules (setting the initial state of the KeepassXC Window to _miniminzed_ in System Settings -> Window Management -> Window Rules
[Plasma 5.12]

1. Set the Window class (application) to Exact Match -> _keepassxc_
![screenshot_20180622_124719](https://user-images.githubusercontent.com/28687075/41772843-785adec4-761a-11e8-9bca-164873acd92e.png)

2. Under the tab _Size & Position_ set Minimized to _Apply Initially --> Yes_
![image](https://user-images.githubusercontent.com/28687075/41772880-8d78bb5a-761a-11e8-97d5-8971f784ad9b.png)

**Comment by railes** [comment]
Hi raphwriter,

thanks for your answer, even I posted this some month ago. But I don't think you 'got the point' what kind of problem I had.

My idea was to open immediately the KeePass database after login and KeePassXC should go then immediately to the background in the tray.
But I learned from phoerious that the behaviour of KeePassXC is not designed like this.

And I must say, that maybe not to open the database on login is the better way. KeyPassXC goes after login automatically in the background (without KDE settings you described, KeyPassXC settings 'Minimize windows at application startup' does this). And when I need KeyPass the first time with CTRL+ALT+Y, it asks for the password and goes back to the background. That's fine for me.

But thanks a lot for your answer!
